setendings.py

```diff
@@ -0,0 +1,624 @@
+    child nodes first, since it is sometimes unset in the last child.
+    Postorder setting of children results in the assumption: if a node has the
+    propert set, then all of its children are set.
+    """
+
+    # Compare with "None" since value could be "0", which is falsey.
+    # Returning a.s.a.p. reduces the runtime (considerably?).
+    if node.end_lineno is not None and node.end_col_offset is not None:
+        return  # since properties are already set.
+
+    # Doing this first makes it a postorder traversal.
+    for child in node.get_children():  # get_children() returns a generator.
+        _set_end_col_offset_by_string(child)
+
+    node_as_string = str(node)
+    node_name = node_as_string[ : node_as_string.find('(')]
+    # print('node: ', node_name)
+
+    #❕TODO: set up the below to work with postorder traversal, setting all
+    # child properties, calling the different transform functions by node
+    # type in nodes.. nodes[node_name]['function']
+
+
+
+
+
+
+    
+
+
+    # Temporary:    
+    # print('props: ', dir(node))
+    last_child_end = node.last_child().end_col_offset
+    if last_child_end is None:
+        logging.error("⛔ ERROR:️ last child's end_col_offset property is None.")
+        # sys.exit(1)
+        raise Exception('''⛔ last child's end_col_offset property is None.''')
+        # TODO: get the type of astroid node (Const?, Expr?, ...)
+        # and calculate its 
+
+    print('last: ', node.last_child().end_col_offset)
+    node.end_col_offset = node.last_child().end_col_offset
+    
+
+
+
+
+def _set_end_col_offset_by_value(node):
+    """Set the end_col_offset property by the value attribute.
+    """
+    # Guard: should have .value property
+    if 'value' not in dir(node):
+        raise Exception('''⛔️ node must have the .value property.''')
+        sys.exit(1)
+
+    # Guard: something with .value shouldn't have children. Use a different
+    # function instead of _set_end_col_offset_by_value.
+    for child in node.get_children():  # get_children() returns a generator.
+        raise Exception('''⛔ _set_end_col_offset_by_value function 
+                        should not be called if has children.''')
+        sys.exit(1)
+    node.end_col_offset = node.col_offset + len(str(node.value))
+
+
+# TODO: maybe separate the following concerns into different files.
+
+# Define the functions to transform the nodes.
+# These functions are registered to their respective node type with the
+# register_transform function.
+# Mutate the nodes, adding the properties: `end_lineno`, `end_col_offset`.
+################################################################################
+def set_general(node):
+    """General function called by many of the other transform functions.
+    Populate ending locations for astroid node."""
+    _set_end_lineno(node)
+    _set_end_col_offset_by_string(node)
+
+#######
+
+def set_arguments(node):
```

Comment:
You don't need to define separate functions. You just need to register_transform with the `set_general` directly.